docs/utilities/middleware_factory.md

```diff
@@ -3,89 +3,144 @@ title: Middleware factory
+    ```
 
-@lambda_handler_decorator
-def middleware_before_after(handler, event, context):
-	# logic_before_handler_execution()
-	response = handler(event, context)
-	# logic_after_handler_execution()
-	return response
+### Middleware with after logic
 
-@middleware_before_after
-def lambda_handler(event, context):
-	...
-```
+=== "getting_started_middleware_after_logic_function.py"
+    ```python hl_lines="7 14 15 21-23 37"
+    --8<-- "examples/middleware_factory/src/getting_started_middleware_after_logic_function.py"
+    ```
 
-## Middleware with params
+=== "getting_started_middleware_after_logic_payload.json"
+
+    ```json
+    --8<-- "examples/middleware_factory/src/getting_started_middleware_after_logic_payload.json"
+    ```
+
+### Middleware with params
 
 You can also have your own keyword arguments after the mandatory arguments.
 
-```python hl_lines="2 12" title="Accepting arbitrary keyword arguments"
-@lambda_handler_decorator
-def obfuscate_sensitive_data(handler, event, context, fields: List = None):
-	# Obfuscate email before calling Lambda handler
-	if fields:
-		for field in fields:
-			if field in event:
-				event[field] = obfuscate(event[field])
+=== "getting_started_middleware_with_params_function.py"
+    ```python hl_lines="6 27 28 29 33 49"
+    --8<-- "examples/middleware_factory/src/getting_started_middleware_with_params_function.py"
+    ```
+
+=== "getting_started_middleware_with_params_payload.json"
 
-	return handler(event, context)
+    ```json hl_lines="18 19 20"
+    --8<-- "examples/middleware_factory/src/getting_started_middleware_with_params_payload.json"
+    ```
+
+## Advanced
+
+For advanced use cases, you can instantiate [Tracer](../core/tracer.md) inside your middleware, and add annotations as well as metadata for additional operational insights.
 
-@obfuscate_sensitive_data(fields=["email"])
-def lambda_handler(event, context):
-	...
-```
+=== "advanced_middleware_tracer_function.py"
+    ```python hl_lines="7 9 12 16 17 19 25 42"
+    --8<-- "examples/middleware_factory/src/advanced_middleware_tracer_function.py"
+    ```
 
-## Tracing middleware execution
+=== "advanced_middleware_tracer_payload.json"
+
+    ```json
+    --8<-- "examples/middleware_factory/src/advanced_middleware_tracer_payload.json"
+    ```
+
+![Middleware advanced Tracer](../media/middleware_factory_tracer_2.png)
+
+### Tracing middleware **execution**
 
 If you are making use of [Tracer](../core/tracer.md), you can trace the execution of your middleware to ease operations.
 
 This makes use of an existing Tracer instance that you may have initialized anywhere in your code.
 
-```python hl_lines="3" title="Tracing custom middlewares with Tracer"
-from aws_lambda_powertools.middleware_factory import lambda_handler_decorator
+???+ warning
+    You must enable Active Tracing in your Lambda function when using this feature, otherwise Lambda will not be able to send traces to XRay..
```

Comment:
```suggestion
    You must [enable Active Tracing](../core/tracer/#permissions) in your Lambda function when using this feature, otherwise Lambda cannot send traces to XRay.
```